src/resources/marketplace.js

```diff
@@ -108,6 +113,46 @@ class Marketplace extends Adaptable {
     })
   }
 
+  async search(searchQuery, listingType, filters = []) {
```

Comment:
style nit: since this is going to be a publicly exposed method used by DApp developers, let's document it well. I have been trying to encourage folks to use [JSDoc](http://usejsdoc.org/about-getting-started.html) format (but if anyone else has a better suggestion, please lmk).